### DISCUSSION TITLE
How to disable data caching in useAsyncData?

### DISCUSSION THREAD

**Discussion by rudimadima** [nuxt/nuxt#16407] [Questions] [answered]
Hi.
I'm trying to build my new project on Nuxt 3. I use `useAsyncData`:
```
const { data, error, pending, refresh } = await useAsyncData(
  `get question ${questionId}}`,
  () => {
    console.log('fetching data...');
    return questionsApi.getOne(questionId); // It just returns promise from $fetch 
  }
);
```
Due to caching of the request response, the data on the page is not updated when I leave it to another page, and then return back.
I tried to add `Math.random()` to the unique key, but it gave an error:
```
[Vue warn]: Hydration node mismatch...
```
How do I disable caching?

**Comment by Diizzayy** [comment, marked as answer]
@rudimadima You can disable the cache by setting `initialCache` to false as seen below

`useAsyncData("key", ()=> {}, { initialCache: false })`

**Reply by ironexdev** [reply]
It would be nice to have this behavior described in the documentation https://v3.nuxtjs.org/api/composables/use-async-data/ - other options are already described there, but initialCache is missing

-- Edit --

Nevermind, just found out about this https://github.com/nuxt/framework/pull/8885 -> initialCache option is being removed.

**Reply by mrtdmrmrt** [reply]
I just encountered the same situation. What do you recommend instead of initialCache?
